presto-main/src/main/java/com/facebook/presto/sql/gen/ByteCodeUtils.java

```
@@ -290,4 +294,27 @@ public static ByteCodeNode generateWrite(CallSiteBinder callSiteBinder, Compiler
                                 .invokeInterface(Type.class, methodName, void.class, BlockBuilder.class, type.getJavaType()))
                         .build());
     }
+
+    public static void checkIdentifier(String name)
```

add the check to the invoke `ByteCodeNode` so we fail fast